Shared/Samples/Set up location-driven geotriggers/SetUpLocationDrivenGeotriggersView.swift

```diff
@@ -0,0 +1,332 @@
+        @Published var hasCurrentSection = false
+        
+        /// A Boolean value indicating whether there is a point-of-interest.
+        @Published var hasPointOfInterest = false
+        
+        /// A Boolean value indicating whether to show an error alert.
+        @Published var isShowingAlert = false
+        
+        /// The error shown in the alert.
+        @Published var error: Error? {
+            didSet { isShowingAlert = error != nil }
+        }
+        
+        /// The handle for the geotrigger monitor notification task.
+        private var notificationTaskHandle: Task<Void, Never>!
+        
+        init() {
+            // Create simulated data source with the route polyline.
+            locationDataSource = SimulatedLocationDataSource(polyline: .walkingTour)
+            
+            // Create location display.
+            locationDisplay = Self.makeLocationDisplay(dataSource: locationDataSource)
+        }
+        
+        deinit {
+            notificationTaskHandle.cancel()
+        }
+        
+        /// Creates geotriggers from the map's operational layers.
+        func startGeotriggerMonitoring() {
+            // Get the service feature tables from the map's operational layers.
+            if let operationalLayers = map.operationalLayers as? [FeatureLayer],
+               let gardenSectionsLayer = operationalLayers.first(where: { $0.item?.id == .gardenSectionsLayer }),
+               let gardenPOIsLayer = operationalLayers.first(where: { $0.item?.id == .gardenPOIsLayer }),
+               let gardenSections = gardenSectionsLayer.featureTable as? ServiceFeatureTable,
+               let gardenPOIs = gardenPOIsLayer.featureTable as? ServiceFeatureTable {
+                // Create geotriggers for each of the service feature tables.
+                let geotriggerFeed = LocationGeotriggerFeed(locationDataSource: locationDataSource)
+                makeGeotrigger(feed: geotriggerFeed, featureTable: gardenSections, bufferDistance: 0.0, fenceGeotriggerName: sectionFenceGeotriggerName)
+                makeGeotrigger(feed: geotriggerFeed, featureTable: gardenPOIs, bufferDistance: 10.0, fenceGeotriggerName: poiFenceGeotriggerName)
+            }
+        }
+        
+        /// Create a geotrigger monitor and start observing its notifications.
+        /// - Parameters:
+        ///   - feed: The `GeotriggerFeed` that is monitored for changes.
+        ///   - featureTable: The `ServiceFeatureTable` that contains the features
+        ///   to use in the `FeatureFenceParameters`.
+        ///   - bufferDistance: The `Double` buffer distance in meters to apply to
+        ///   the features when checking if an `FenceGeotrigger` condition is met.
+        ///   - fenceGeotriggerName: The name for the `FenceGeotrigger`.
+        private func makeGeotrigger(feed: GeotriggerFeed, featureTable: ServiceFeatureTable, bufferDistance: Double, fenceGeotriggerName: String) {
```

Comment:
A few improvements to this method:

1. `makeGeotrigger` implies it is a [factory method](https://www.swift.org/documentation/api-design-guidelines#strive-for-fluent-usage). But this method is not returning anything. A possible change is `makeGeotriggerMonitor` and let it return a monitor object. Then in the outer driver method (`startGeotriggerMonitoring`, which may have a better name), add the created monitor objects to the array.
2. Starting the monitors can also be done in the driver method, so this method is free from async context.
3. Currently there is only 1 `notificationTaskHandle` in the data model, but multiple async sequences are monitored. We can discuss this more thoroughly. Please keep an eye on the discussion in the swift-general channel.
    - Use a Task block to wrap a task group of Void.self, then add all the sequences to the group
    - Use 1 Task block for each sequence, and store the block handles in an array
    - Use multiple task modifiers